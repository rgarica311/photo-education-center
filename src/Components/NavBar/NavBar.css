.nav-link:hover {
  color: red;
}

.nav-link {
  color: lightgrey;
  text-decoration: none;
  font-family: 'Work Sans', sans-serif;
  font-weight: 700;
  font-size: .8rem;
  position: relative;
  height: 100%;
  display: flex;
  align-items: center;
}

.list-container {
  background-color: grey;
  position: fixed;
  top: 80px;
  margin: 0;
  padding: 0;
  width: 350px;
  height: max-content;
  z-index: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

#about {
  left: 53vw;
}

#classes {
  left: 73vw;
}

#adventures {
  left: 53vw;
}

#events {
  left: 73vw;
}

.dd-menu-el {
  text-decoration: none;
  color: lightgrey;
  list-style: none;
  padding: 10px;
  border-bottom: 1px solid black;
  display: flex;
  align-items: center;
  min-height: 50px;
}

.dd-menu-el:hover {
  background-color: darkgrey;
  cursor: pointer;
}

.header-submenu-container {
  display: flex;
  flex-direction: column;
}

a {
  text-decoration: none;
  color: lightgrey;
}

a:hover {
  color: black;
}

@media screen and (max-width: 812px) {
  .logo{
    font-size: .7rem;
    margin-bottom: 0px;
  }

  .nav-link {
    font-size: .4rem;
    height: 0%;
    align-items: baseline;
  }
}

@media screen and (max-width: 600px) {
  .logo {
    font-size: .5rem;
    top: 1.78px;
    position: relative;
  }

  .nav-link {
    font-size: .5rem;
  }
}
