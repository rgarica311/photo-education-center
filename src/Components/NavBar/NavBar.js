import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './NavBar.css';

console.log( 'link', Link );

const headerStyle = {
  position: 'fixed',
  top: '0',
  left: '0',
  backgroundColor: 'white',
  display: 'flex',
  height: '80px',
  zIndex: '2'
};

const navStyle = {
  display: 'flex',
  textTransform: 'uppercase',
  justifyContent: 'space-evenly',
  width: '100vw',
  alignItems: 'center'
};

const homelinkStyle = {
  color: 'black',
  textDecoration: 'none',
  fontSize: '1rem',
  fontFamily: 'Raleway',
  fontWeight: '900'
};

export default class NavBar extends Component {
  constructor( props ) {
    super( props );
    this.state = {
      topNav: props.topNav,
      aboutMenu: props.aboutMenu,
      classesMenu: props.classesMenu,
      hover: false
    };
  }

  handleHover(e) {
    console.log('e.target', e.target.text)
    this.setState({
      hover: true,
      target: e.target.text
    });
  }

  handleMouseEnterUl(e) {
    this.setState({
      hover: true,
    })
  }

  handleMouseLeave(){
    this.setState({
      hover: false
    })
  }

  renderMenu() {
    Object.keys(this.state.aboutMenu).forEach((key) => {
      console.log(`this.state.aboutMenu${key}: ${this.state.aboutMenu[key]}, ${typeof(this.state.aboutMenu[key])}`)
    })
    const aboutMenu = this.state.aboutMenu;
    const classesMenu = this.state.classesMenu;
    return (
      <>
      {
        this.state.target === 'About'
          ? <ul onMouseLeave={e => this.handleMouseLeave( e )} onMouseEnter={e => this.handleMouseEnterUl( e )} id='about' className='list-container'>
            {
              Object.keys(aboutMenu).map(key => {
                if(aboutMenu[key] !== null && typeof(aboutMenu[key]) !== 'number'){
                  return <Link key={aboutMenu[key]} to={`/${aboutMenu[key].split(' ').join('-')}`}><li className='dd-menu-el' key={aboutMenu[key]}>{aboutMenu[key]}</li></Link>
                }
                return null;
              })
            }
          </ul>
          : this.state.target === 'Classes'
              ? <ul onMouseLeave={e => this.handleMouseLeave( e )} onMouseEnter={e => this.handleMouseEnterUl( e )} id='classes' className='list-container'>
                {
                  Object.keys(classesMenu).map(key => {
                    if(classesMenu[key] !== null && typeof(classesMenu[key]) !== 'number'){
                      return <Link key={classesMenu[key]} to={classesMenu[key].split(' ').join('-')}><li className='dd-menu-el' key={classesMenu[key].value}>{classesMenu[key]}</li></Link>
                    }
                    return null;
                  })
                }
              </ul>
              /*: this.state.target === 'Adventures'
                  ? <ul onMouseLeave={e => this.handleMouseLeave( e )} onMouseEnter={e => this.handleMouseEnterUl( e )} id='adventures' className='list-container'>
                    {
                      Object.keys(adventuresMenu).map(key => {
                        if(adventuresMenu[key] !== null && typeof(adventuresMenu[key]) !== 'number'){
                          return <Link key={adventuresMenu[key]} to={adventuresMenu[key].split(' ').join('-')}><li className='dd-menu-el' key={adventuresMenu[key]}>{adventuresMenu[key]}</li></Link>
                        }
                      })
                    }
                  </ul>
                  : this.state.target === 'Events'
                      ? <ul onMouseLeave={e => this.handleMouseLeave( e )} onMouseEnter={e => this.handleMouseEnterUl( e )} id='events' className='list-container'>
                          {
                            Object.keys(eventsMenu).map(key => {
                              if(eventsMenu[key] !== null && typeof(eventsMenu[key]) !== 'number'){
                                return <Link key={eventsMenu[key]} to={eventsMenu[key].split(' ').join('-')}><li className='dd-menu-el' key={eventsMenu[key]}>{eventsMenu[key]}</li></Link>
                              }
                            })
                          }
                        </ul>*/
                      : null
      }
      </>

    );
  }

  render() {
    return (
      <>
        <header style={headerStyle} className='nav-bar'>
          <nav style={navStyle}>
            <Link style={homelinkStyle} to='/'><h1 className='logo'>
              PHOTO EDUCATION CENTER
            </h1>
            </Link>
            <Link onMouseLeave={e => this.handleMouseLeave( e )} onMouseOver={e => this.handleHover( e )} className='nav-link' to='/' >{this.state.topNav.col1}</Link>
            <Link onMouseLeave={e => this.handleMouseLeave( e )} onMouseOver={e => this.handleHover( e )} className='nav-link' to='/beginner'>{this.state.topNav.col2}</Link>
            {/*<Link onMouseLeave={e => this.handleMouseLeave( e )} onMouseOver={e => this.handleHover( e )} className='nav-link' to='/cuba 2020'>{this.state.topNav.col3}</Link>*/}
            {/*<Link onMouseLeave={e => this.handleMouseLeave( e )} onMouseOver={e => this.handleHover( e )} className='nav-link' to='/ready, set, nikon, shoot!'>{this.state.topNav.col4}</Link>*/}
          </nav>
        </header>
        {this.state.hover === true ? this.renderMenu() : null}
      </>
    );
  }

}
