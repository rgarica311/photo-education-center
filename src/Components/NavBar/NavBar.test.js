import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavBar from './NavBar';

describe(`NavBar Component`, () => {
  const topNav = {
    col1: 'test link',
    col2: 'test link'
  };
  it('renders NavBar by default', () => {
    const wrapper = shallow(<NavBar topNav={topNav} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
