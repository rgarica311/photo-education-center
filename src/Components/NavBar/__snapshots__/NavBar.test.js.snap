// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component renders NavBar by default 1`] = `
<Fragment>
  <header
    className="nav-bar"
    style={
      Object {
        "backgroundColor": "white",
        "display": "flex",
        "height": "80px",
        "left": "0",
        "position": "fixed",
        "top": "0",
        "zIndex": "2",
      }
    }
  >
    <nav
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-evenly",
          "textTransform": "uppercase",
          "width": "100vw",
        }
      }
    >
      <Link
        style={
          Object {
            "color": "black",
            "fontFamily": "Raleway",
            "fontSize": "1rem",
            "fontWeight": "900",
            "textDecoration": "none",
          }
        }
        to="/"
      >
        <h1
          className="logo"
        >
          PHOTO EDUCATION CENTER
        </h1>
      </Link>
      <Link
        className="nav-link"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        to="/"
      >
        test link
      </Link>
      <Link
        className="nav-link"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        to="/beginner"
      >
        test link
      </Link>
    </nav>
  </header>
</Fragment>
`;
